src/main/java/org/apache/commons/io/FilenameUtils.java

```diff
@@ -669,7 +669,7 @@ public static String getFullPath(final String fileName) {
      * ~            --&gt; ~
      * ~/           --&gt; ~
      * ~user        --&gt; ~user
-     * ~user/       --&gt; ~user
+     * ~user/       --&gt; ~user/
```

Comment:
This causes the Javadoc to contradict itself with the above's `excluding the final directory separator`.
The question is whether Windows's `"C:\"` unchanged output is a special case or a bug.

Reply:
The existing javadoc and API is very unclear. I do not understand "Gets the full path from a full fileName, which is the prefix + path, and also excluding the final directory separator."

The argument is not a filename. There is no such thing as a "full path" in normal parlance. Bug filed. https://issues.apache.org/jira/browse/IO-836

I also agree that if this method turns ~user/ into ~user/ that's a bug. It should produce ~user.

C:\ should also (80% certainty) turn into C:, not C:\

I further expect there are additional bugs in this method involving Unix shell conventions and platform dependencies, but I'd have to dig into the rest of the class to understand exactly what those are.

What a tasy can of worms you've opened. :-)